core/server/master/src/main/java/alluxio/master/block/DefaultBlockMaster.java

```diff
@@ -831,7 +845,9 @@ public void workerRegister(long workerId, List<String> storageTiers,
   @Override
   public Command workerHeartbeat(long workerId, Map<String, Long> capacityBytesOnTiers,
       Map<String, Long> usedBytesOnTiers, List<Long> removedBlockIds,
-      Map<String, List<Long>> addedBlocksOnTiers, List<Metric> metrics) {
+      Map<String, List<Long>> addedBlocksOnTiers,
+      Map<String, List<String>> removedStorageOnTiers,
```

Comment:
Does the command work after master is restarted? We might need to report all existing lost storage paths when worker is registered with a new master node.

Reply:
Previously we record failed to initialize paths and report via worker register and report failed in storage check paths in worker heartbeat which cannot survive through restart.

Now we record all the lost storage paths(failed to initialize and failed in periodical storage check) in the worker and report them in the worker register. When lost storage find in storage check, we report them via worker heartbeat and also record them in the worker to make this command work after master is restarted.